### PR TITLE
cloudapi: expose upload status

### PR DESCRIPTION
**Pull request by chloenayon**
Expose a more detailed job status result - specifically, include upload status alongside
image status. Expand openapi.yml accordingly and add an UploadStatus field to the 
OSBuildJobResult struct. At the moment, only represent the "success" and "failure" states 
of UploadStatus - to differentiate between "pending" and "running" would involve significant 
design decisions and should be addressed in a separate commit.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change

### CODE REVIEW

**Comment by Gundersanne** [comment]
Looks good! 

The test failures are due to the status response having changed, needs a tweak here https://github.com/osbuild/osbuild-composer/blob/01c21b5a47dc6d96e7a81b204902d904790e5e91/test/cases/api.sh#L148 at least.

File: test/cases/api.sh [hunk 1]
```diff
@@ -145,7 +145,7 @@ do
     --cert /etc/osbuild-composer/client-crt.pem \
     https://localhost/api/composer/v1/compose/"$COMPOSE_ID")
 
-  COMPOSE_STATUS=$(echo "$OUTPUT" | jq -r '.status')
+  COMPOSE_STATUS=$(echo "$OUTPUT" | jq -r '.image_status.status')
 
   if [[ "$COMPOSE_STATUS" != "pending" && "$COMPOSE_STATUS" != "running" ]]; then
     test "$COMPOSE_STATUS" = "success"
```
Comment (Gundersanne) [inline test/cases/api.sh:151]: We could tes tthe upload status by adding a test here by extracting `upload_status` from the COMPOSE_STATUS here and see if it's `success` too.

**Review by Gundersanne — COMMENTED** [review]
The idea looks good, I'd add a test though. It should be easy enough to expand our current api test to include checking the upload_status.

**Review by Gundersanne — APPROVED** [review]
The only failed test now is `fedora_33-x86_64-fedora_iot_commit-boot.json`, unrelated to the PR. Also it succeeded in pipeline #7 which also ran on commit `aa355f117a38c8d22222fb53bff37834601eb285`.

I'd say we merge this.

**Comment by larskarlitski** [comment]
It looks to me like the failure is known and tracked in https://github.com/osbuild/osbuild-composer/issues/1196. Feel free to merge,

**Comment by thozza** [comment]
Hey.

I noticed that after this PR, there is an unused component in the cloudapi scheme:
https://github.com/osbuild/osbuild-composer/blob/d686abfffbb4d805aa55ef3a27673ad7561e3849/internal/cloudapi/openapi.yml#L119-L124
This has been previously not really used, but I thought that eventually we will return more information related to the upload status, than just the type of the upload. E.g for AWS it would be the `ami_id`, for GCP this would be `project_id` and `image_name`. I expect that we will have to return more granular information, which should be presented to the user, so that they can identify the image and use it with the cloud provider. Isn't this change a potential step away from that direction?

**Pull request merged**